### PR TITLE
Replace DistributedContext.getIterator with getEntries.

### PR DESCRIPTION
**Pull request by songy23**
See https://github.com/open-telemetry/opentelemetry-java/issues/401 and https://github.com/open-telemetry/opentelemetry-java/pull/429.

### CODE REVIEW

File: specification/distributedcontext-api.md [hunk 1]
```diff
@@ -140,11 +140,10 @@ Close Scope 1
 
 # DistributedContext
 
-## GetIterator
+## GetEntries
 
-Returns an iterator over the entries in this `DistributedContext`.
-The order of entries is not significant. The iterator MUST not modify the underlying collection of
-entries in this `DistributedContext`.
+Returns the entries in this `DistributedContext`.
```
Comment (bogdandrutu) [inline specification/distributedcontext-api.md]: I would rephrase this to be a bit more generic. What about this: "Based on the language specification this can an immutable collection or an immutable iterator to the collection of entries in this DistributedContext"?

File: specification/distributedcontext-api.md [hunk 2]
```diff
@@ -140,11 +140,10 @@ Close Scope 1
 
 # DistributedContext
 
-## GetIterator
```
Comment (SergeyKanzhelev) [inline specification/distributedcontext-api.md]: as a minimum we need to add an entry to the CHANGELOG.md to track any changes we are making in API description. I understand this change may not worth an RFC

Comment (bogdandrutu) [reply specification/distributedcontext-api.md]: Agree that this change does not require an RFC because does not change the API functionality, it just changes how the data are returned.

**Comment by bogdandrutu** [comment]
@songy23 also please make sure we do not merge this for at least 24h to give everyone a chance to comment if they have concerns.

File: specification/distributedcontext-api.md (hunk 2 above)
Comment (songy23) [reply specification/distributedcontext-api.md]: We don't have a changelog in this repo for now, I added it in https://github.com/open-telemetry/opentelemetry-specification/pull/161.

File: specification/distributedcontext-api.md (hunk 1 above)
Comment (songy23) [reply specification/distributedcontext-api.md]: SG, updated. (I kept the API name as `GetEntries`.)

File: specification/distributedcontext-api.md (hunk 2 above)
Comment (iredelmeier) [reply specification/distributedcontext-api.md]: Would this then count as something for 0.1.0 or 0.2.0?

Comment (songy23) [reply specification/distributedcontext-api.md]: I think it will be in 0.2.0. I'll add it to changelog once #161 is merged.

**Comment by bogdandrutu** [comment]
@open-telemetry/specs-approvers can we merge this or we need more time to comment? I think this is a small change that does not change the functionality.

**Comment by songy23** [comment]
This PR is pending for two days and I haven't seen more comments/concerns. I think it's good to go? :)

**Comment by bogdandrutu** [comment]
@SergeyKanzhelev are you ok with the change?

**Comment by songy23** [comment]
Please merge :)
